### DISCUSSION TITLE
[General] <ERC721Bridge contract>

### DISCUSSION THREAD

**Discussion by TralaTheoKim** [zkSync-Community-Hub/zksync-developers#502] [General]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

I am implementing the ERC721Bridge contract by referring to matterlabs' era-contracts ERC20Bridge contract.

### What is the specific issue or error you're encountering?

Separately, I looked at the Deposit function of the erc20Bridge contract, customized it, modified it to fit erc721, and tested it on the Sepolia network, and the transaction failed.

### Can you share the error messages or logs you're receiving, if any?

Error during deposit :  Error: transaction failed [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (transactionHash="0x3129265cdb5ecf74bc91d8d98adafd7c1775971137339ef827c60f183b6bb3bc", transaction={"hash":"0x3129265cdb5ecf74bc91d8d98adafd7c1775971137339ef827c60f183b6bb3bc","type":2,"accessList":[],"blockHash":null,"blockNumber":null,"transactionIndex":null,"confirmations":0,"from":"0x86c2a1E1DaB458AFc74C809840bBC9a54Ce0719b","gasPrice":{"type":"BigNumber","hex":"0x989b4c95"},"maxPriorityFeePerGas":{"type":"BigNumber","hex":"0x59682f00"},"maxFeePerGas":{"type":"BigNumber","hex":"0x989b4c95"},"gasLimit":{"type":"BigNumber","hex":"0x07a120"},"to":"0x55dC9bEDB73b9a3AaE959dCE99fdc53002B0de46","value":{"type":"BigNumber","hex":"0x00"},"nonce":55,"data":"0x6e553f65000000000000000000000000000000000000000000000000000000000000000100000000000000000000000086c2a1e1dab458afc74c809840bbc9a54ce0719b","r":"0x99381cd9482bb5c8b7d76608b02191ab8cf9ac61734805d7b4c91d727ddc4957","s":"0x4f3398c75dae3538815961532cc0311b9ae813f0051edb1a0106dc64f4eb06b0","v":0,"creates":null,"chainId":11155111}, receipt={"to":"0x55dC9bEDB73b9a3AaE959dCE99fdc53002B0de46","from":"0x86c2a1E1DaB458AFc74C809840bBC9a54Ce0719b","contractAddress":null,"transactionIndex":100,"gasUsed":{"type":"BigNumber","hex":"0xab15"},"logsBloom":"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000","blockHash":"0x3d4a4023b65dfc2de1c589b642db58761cf97d1264b86f164ebc0d1a9e712c0d","transactionHash":"0x3129265cdb5ecf74bc91d8d98adafd7c1775971137339ef827c60f183b6bb3bc","logs":[],"blockNumber":5820490,"confirmations":1,"cumulativeGasUsed":{"type":"BigNumber","hex":"0xa214eb"},"effectiveGasPrice":{"type":"BigNumber","hex":"0x989b4c95"},"status":0,"type":2,"byzantium":true}, code=CALL_EXCEPTION, version=providers/5.7.2)
    at Logger.makeError (/Users/gimgidae/go/src/Hardhat/ethereum/node_modules/@ethersproject/logger/src.ts/index.ts:269:28)
    at Logger.throwError (/Users/gimgidae/go/src/Hardhat/ethereum/node_modules/@ethersproject/logger/src.ts/index.ts:281:20)
    at EthersProviderWrapper.<anonymous> (/Users/gimgidae/go/src/Hardhat/ethereum/node_modules/@ethersproject/providers/src.ts/base-provider.ts:1549:24)
    at step (/Users/gimgidae/go/src/Hardhat/ethereum/node_modules/@ethersproject/providers/lib/base-provider.js:48:23)
    at Object.next (/Users/gimgidae/go/src/Hardhat/ethereum/node_modules/@ethersproject/providers/lib/base-provider.js:29:53)
    at fulfilled (/Users/gimgidae/go/src/Hardhat/ethereum/node_modules/@ethersproject/providers/lib/base-provider.js:20:58) {
  reason: 'transaction failed',

### Have you made any recent changes to the contract before encountering this issue?

doesn't exist.

### Are there any external libraries or contracts that your contract interacts with?

I used @matterlabs/zksync-contracts and @openzeppelin/contracts.

### Can you provide the relevant portions of your contract code where the issue is occurring?

    function deposit(uint256 tokenId, address toL2) public nonReentrant {
        token.safeTransferFrom(msg.sender, address(this), tokenId);
        bytes memory data = abi.encodeWithSelector(
            IL2ERC721Bridge.finalizeDeposit.selector,
            msg.sender,
            toL2,
            tokenId
        );

        bytes32 l2TxHash = zkSyncMailbox.requestL2Transaction(
            l2BridgeAddress,
            0,
            data,
            500000,
            0,
            new bytes[](0),
            msg.sender
        );

        emit DepositInitiated(tokenId, msg.sender, toL2, l2TxHash);
    }

### Have you tried to isolate the problem, and if so, what were the results?

Additionally, we are looking into the cause of the problem and looking into the zkSync consensus.

### What steps have you already taken to try to resolve the issue?

see above

### Additional Details

I would like to implement the ERC721Bridge contract. Can you provide additional reference and helpful information?

**Comment by bxpana** [comment]
Will check this out and get back to you

**Reply by bxpana** [reply]
@TralaTheoKim can you provide a link to the repo so we can test on our end to see what's up?

**Reply by TralaTheoKim** [reply]
https://github.com/TralaTheoKim/zkSync_test/blob/develop/zkSync/contracts/L1ERC721Bridge.sol
Are you able to access the GitHub link above?

**Reply by bxpana** [reply]
Yes I can. Will take a look this week

**Reply by bxpana** [reply]
For the ERC20 bridge when you bridge a new token, a contract is deployed on L2, and tokens are minted. This might be something you need to add to your bridge where a new ERC721 contract is deployed on L2 and the NFT is minted on L2. Do you have that included?
